src/zarrman/path.rs

```diff
@@ -29,6 +29,9 @@ impl ReqPath {
             prefix.push(&zarr_id);
             return Some(ReqPath::Dir(prefix));
         };
+        let Some(checksum) = checksum.strip_suffix(".zarr") else {
```

Comment:
FWIW: in effect we would accept/handle both URLs with `.zarr` and without alike, but we would show (code below) them as `.zarr`. Correct?  
I just wonder if better would be to make it without a hidden side-effect and apply only if extension is `.zarr`? (may be later we would like to add some special handling for whenever there is no `.zarr`)